### PR TITLE
out_cloudwatch_logs: add record_accessor support for log group and st…

### PR DESCRIPTION
**Pull request by PettitWesley**
…ream name templating

Signed-off-by: Wesley Pettit <wppttt@amazon.com>

### CODE REVIEW

File: pipeline/outputs/cloudwatch.md [hunk 1]
```diff
@@ -18,8 +18,10 @@ See [here](https://github.com/fluent/fluent-bit-docs/tree/43c4fe134611da471e706b
...
+| log_group_template   | Template for Log Group name using Fluent Bit [record_accessor](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/record-accessor) syntax. If the template translation fails an error is logged and the log_stream_name or log_stream_prefix are used instead. See the tutorial below for an example.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
```
Comment (krispraws) [inline pipeline/outputs/cloudwatch.md:21]: Should the documentation mention that the field is optional, and that it overrides the log_group_name if present?
Nit: Missing comma after `If the template translation fails`

File: pipeline/outputs/cloudwatch.md [hunk 2]
```diff
@@ -18,8 +18,10 @@ See [here](https://github.com/fluent/fluent-bit-docs/tree/43c4fe134611da471e706b
...
+| log_stream_template   | Template for Log Stream name using Fluent Bit [record_accessor](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/record-accessor) syntax. If the template translation fails an error is logged and the log_stream_name or log_stream_prefix are used instead. See the tutorial below for an example.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
```
Comment (krispraws) [inline pipeline/outputs/cloudwatch.md:24]: Same comment as above

File: pipeline/outputs/cloudwatch.md [hunk 3]
```diff
@@ -95,6 +97,57 @@ Example:
 
 If you enable a single worker, you are enabling a dedicated thread for your CloudWatch output. We recommend starting without workers, evaluating the performance, and then enabling a worker if needed. For most users, the plugin can provide sufficient throughput without workers.
 
+### Log Stream and Group Name templating using record_accessor syntax
+
+Sometimes, you may want the log group or stream name to be based on the contents of the log record itself. This plugin supports templating log group and stream names using Fluent Bit [record_accessor](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/record-accessor) syntax. 
+
+Here is an example usage, for a common use case- templating log group and stream names based on Kubernetes metadata. 
+
+Recall that the kubernetes filter can add metadata if you run Fluent Bit in kubernetes that will look like the following:
```
Comment (krispraws) [inline pipeline/outputs/cloudwatch.md:106]: Nit: punctuation needed

**Comment by patrick-stephens** [comment]
@PettitWesley can we get this updated and merged? A user had an issue in Slack that was resolved after finding this in the code so be good to document it.

**Comment by PettitWesley** [comment]
@patrick-stephens thanks for reminding me, had forgotten about this one

**Pull request merged**